src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.health >= 50
+      && animal.hidden === false) {
+      animal.health -= 50;
+
+      Animal.alive = Animal.alive.filter(a => a.health !== 0);
```

Comment:
```suggestion
      if (animal.health === 0) {
        Animal.alive = Animal.alive.filter(a => a.health !== 0);
      }
```
It is better to check it here to avoid redundant method calls

Reply:
Hello. I did that from the beginning. But then reworked to remove that if. Because these ifs are constantly given comments